Src/FluentAssertions/Collections/GenericCollectionAssertions.cs

```diff
@@ -944,6 +944,98 @@ public AndConstraint<TAssertions> ContainInOrder(params T[] expected)
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    /// <exception cref="ArgumentNullException"><paramref name="expected"/> is <c>null</c>.</exception>
+    public AndConstraint<TAssertions> ContainInConsecutiveOrder(IEnumerable<T> expected, string because = "",
+        params object[] becauseArgs)
+    {
+        Guard.ThrowIfArgumentIsNull(expected, nameof(expected), "Cannot verify ordered containment against a <null> collection.");
+
+        bool success = Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .ForCondition(Subject is not null)
+            .FailWith("Expected {context:collection} to contain {0} in order{reason}, but found <null>.", expected);
+
+        if (success)
+        {
+            IList<T> expectedItems = expected.ConvertOrCastToList();
+            IList<T> actualItems = Subject.ConvertOrCastToList();
+
+            int highestIndex = -1;
+            int index;
+
+            Func<T, T, bool> areSameOrEqual = ObjectExtensions.GetComparer<T>();
+            int expectedItemsCount = expectedItems.Count;
+
+            for (index = 0; index < expectedItemsCount; index++)
+            {
+                T expectedItem = expectedItems[index];
+                if (index == 0)
+                {
+                    actualItems = actualItems.SkipWhile(actualItem => !areSameOrEqual(actualItem, expectedItem)).ToArray();
+                    if (actualItems.Count >= expectedItemsCount - 1)
+                    {
+                        if (actualItems.Take(expectedItemsCount - 1).Equals(expectedItems))
+                        {
+                            return new AndConstraint<TAssertions>((TAssertions)this);
+                        }
+                    }
+                }
+                else
+                {
+                    if (actualItems.Any() && areSameOrEqual(actualItems.First(), expectedItem))
+                    {
+                        highestIndex = index;
+                    }
+                    else
+                    {
+                        actualItems = actualItems.SkipWhile(actualItem => !areSameOrEqual(actualItem, expectedItems[0])).ToArray();
+                        index = 0;
+                    }
+                }
+
+                if (actualItems.Any())
+                {
+                    actualItems = actualItems.Skip(1).ToArray();
+                }
+                else
+                {
+                    Execute.Assertion
+                    .BecauseOf(because, becauseArgs)
+                    .FailWith(
+                        "Expected {context:collection} {0} to contain items {1} in order{reason}" +
+                        ", but {2} (index {3}) did not appear (in the right order).",
+                        Subject, expected, expectedItem, highestIndex + 1);
```

Comment:
🔍 nitpick: missing indent

```suggestion
                        .BecauseOf(because, becauseArgs)
                        .FailWith(
                            "Expected {context:collection} {0} to contain items {1} in order{reason}" +
                            ", but {2} (index {3}) did not appear (in the right order).",
                            Subject, expected, expectedItem, highestIndex + 1);
```